### PR TITLE
Update Readme with latest changes

### PR DESCRIPTION
**Pull request by sridhargaddam**
This patch updates the documentation with some recent changes
along with the submariner architecture diagram.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,28 +41,28 @@ Note that Submariner is in the <strong>pre-alpha</strong> stage, and should not
...
+The `submariner-gateway` pods are run on the gateway labelled nodes, and will perform leader election between them to elect an active IPsec endpoint. `submariner-route-agent` as a component is run on every node, and is aware of the current leader. While running on the gateway node that is the leader, it will create a VxLAN VTEP interface to which all the worker nodes are connected, whereas on the non-leader nodes, it will setup a VXLAN tunnel with remote VTEP as the active gateway node, configures routes, programs necessary iptable rules and enables full connectivity to the remote clusters. The MTU of Submariner VxLAN tunnel is configured based on the MTU of the default interface on the host minus VxLAN overhead.
```
Comment (tpantelis) [inline README.md]:
```suggestion
The `submariner-gateway` pods are run on the gateway-labeled nodes and perform a leader election between them to elect an active IPsec endpoint. The `submariner-route-agent` component runs on every node and is aware of the current active gateway. When running on the active gateway node, the `submariner-route-agent` creates a VxLAN VTEP interface to which the `submariner-route-agent` instances running on the other worker nodes connect by establishing a VXLAN tunnel with the VTEP of the active gateway node. The `submariner-route-agent` also configures routes and programs the necessary iptable rules to enable full connectivity to the remote clusters. The MTU of Submariner VxLAN tunnel is configured based on the MTU of the default interface on the host minus VxLAN overhead.
```

File: README.md [hunk 2]
```diff
@@ -41,28 +41,28 @@ Note that Submariner is in the <strong>pre-alpha</strong> stage, and should not
...
+Upon startup, the `submariner-gateway` pod that is elected leader will perform a reconciliation process that ensures it is the sole endpoint for this cluster. This is part of the reason why it is important to have unique cluster IDs between clusters, as two clusters with the same ID will reconcile each other out of existence.
 
-Upon failure, another Submariner pod (on one of the other gateway hosts) will gain leadership and perform reconciliation to ensure it is the active leader. When done, the remote clusters will reconcile the IPsec endpoint to the new endpoint, and connection will be re-established. In the interim, the `submariner-route-agent` pods will update the route tables on each node to point towards the new endpoint host.
+Upon failure, another `submariner-gateway` pod (on one of the other gateway hosts) will gain leadership and perform reconciliation to ensure it is the active leader. When done, the remote clusters will reconcile the IPsec endpoint to the new endpoint, and connection will be re-established. In the interim, the `submariner-route-agent` pods will update the route tables on each node to point towards the new endpoint host.
```
Comment (tpantelis) [inline README.md]:
```suggestion
Upon failure, another `submariner-gateway` pod (on one of the other gateway nodes) will gain leadership and perform reconciliation to ensure it is the active leader. When done, the remote clusters will reconcile the IPsec endpoint to the new endpoint, and connection will be re-established. In addition, the `submariner-route-agent` pods will update the route tables on each node to point to the new active gateway node.
```

File: README.md [hunk 3]
```diff
@@ -41,28 +41,28 @@ Note that Submariner is in the <strong>pre-alpha</strong> stage, and should not
...
 
-## submariner
+## submariner-gateway
 
-submariner (compiled as the binary `submariner-engine`) has a few controllers built into it that establish state. It is responsible for running/interfacing with Charon to establish IPsec tunnels, as well as updating local cluster information into the central broker to share information between clusters. 
+submariner-gateway (compiled as the binary `submariner-engine`) has a few controllers built into it that establish state. It is responsible for running/interfacing with Charon to establish IPsec tunnels, as well as updating local cluster information into the central broker to share information between clusters.
 
-submariner-engine runs and utilizes leader election to establish an active gateway node, which is used to facilitate IPsec tunnel connections to remote clusters. It also manipulates the IPtables rules on each node to enable a. forwarding of traffic and b. SNAT for local node traffic.
+submariner-gateway runs as a DaemonSet on the gateway labelled nodes and utilizes leader election to establish an active gateway node, which is used to facilitate IPsec tunnel connections to remote clusters.
```
Comment (tpantelis) [inline README.md]:
```suggestion
submariner-gateway runs as a DaemonSet on the gateway-labeled nodes and utilizes leader election to establish an active gateway node, which is used to facilitate IPsec tunnel connections to remote clusters.
```

File: README.md (hunk 1 above)
Comment (sridhargaddam) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (sridhargaddam) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (sridhargaddam) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -72,13 +72,13 @@ The submariner-route-agent runs as a DaemonSet on all Kubernetes nodes, and ensu
...
 
 # Prerequisites
 
 Submariner has a few requirements in order to get started:
 
-- At least 3 Kubernetes clusters, one of which is designated to serve as the central broker that is accessible by all of your connected clusters; this can be one of your connected clusters, but comes with the limitation that the cluster is required to be up in order to facilitate interconnectivity/negotiation
+- At least 2 Kubernetes clusters, one of which is designated to serve as the central broker that is accessible by all of your connected clusters; this can be one of your connected clusters, but comes with the limitation that the cluster is required to be up in order to facilitate interconnectivity/negotiation
 - Different cluster/service CIDR's (as well as different kubernetes DNS suffixes) between clusters. This is to prevent traffic selector/policy/routing conflicts.
```
Comment (mangelajo) [inline README.md]: I would also add, - Worker node IPs on all cluster must be outside of the cluster/service CIDR ranges.

Comment (sridhargaddam) [reply README.md]: Sure, I will add it. Thanks for reviewing the PR @mangelajo

**Pull request merged**